src/Microsoft.AspNetCore.Authentication/ClaimsTransformationMiddleware.cs

```diff
@@ -39,13 +39,18 @@ public class ClaimsTransformationMiddleware
             handler.RegisterAuthenticationHandler(context.GetAuthentication());
             try
             {
-                if (Options.Transformer != null)
+                var transform = Options.Transformer;
+                if (transform == null && Options.TransformerType != null)
+                {
+                    transform = context.RequestServices.GetService(Options.TransformerType) as IClaimsTransformer;
```

Comment:
Eureka :smile:

Alternatively, a warning message when `GetService() == null` could do the job too, if you're not a huge fan of `GetRequiredService`.